IdentityCore/src/cache/crypto/MSIDWPJKeyPairWithCert.h

```diff
@@ -0,0 +1,46 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+
+#import <Foundation/Foundation.h>
+#import "MSIDKeychainIdentity.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MSIDWPJKeyPairWithCert : NSObject <MSIDKeychainIdentity>
+{
+    SecCertificateRef _certificateRef;
+    NSData *_certificateData;
+    NSString *_certificateSubject;
+    NSString *_certificateIssuer;
+    SecKeyRef _privateKeyRef;
+}
+
+@property (nonatomic, readonly) SecKeyRef privateKeyRef;
+@property (nonatomic, readonly) SecKeyRef publicKeyRef;
+@property (nonatomic, readonly) NSDictionary *privateKeyDict;
+
```

Comment:
From the code I can see that privateKeyDict is the result of SecItemCopyMatching for privateKey using query dictionary. Is this privateKeyDict used for any purpose since we have the privateKeyRef?

Reply:
I was initially trying to keep same interface for both MSIDWPJKeyPairWithCert and MSIDAssymetricKeyPairWithCert, and privateKeyDict was needed there (since it's used for POP in other places). I now actually decided to split and simplify WPJ-related class, and keep MSIDAssymetricKeyPairWithCert as is and separate for now. Please take another look.

Reply:
Resolving since privateKeyDict has been removed.